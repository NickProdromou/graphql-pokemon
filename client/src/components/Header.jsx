import React from "react";
import Styled from "styled-components";
import { Link } from 'react-router-dom';
import { Page, Row, Column } from "hedron";
import { spacing, colours } from "../style/variables";
import { type } from "../style/mixins/index";

const Header = () => (
  <StyledHeader className="Header">
    <Page>
      <Row>
        <Column>
          <Link className="HeaderLink" to="/">
            <h1 className="HeaderTitle">Compare a pok√©mon</h1>
          </Link>
        </Column>
      </Row>
    </Page>
  </StyledHeader>
);

const StyledHeader = Styled.header`
  background: ${colours.primaryColorDark};
  box-shadow: 1px 4px 17px rgba(0,0,0,0.5);

  .HeaderLink {
    text-decoration: none;
  }

  .HeaderTitle {
    ${type("heading2")};
    color: ${colours.primaryTextColor};    
    font-weight: bold;
    padding: 0 ${spacing.mid.level1}; 
  }
`;

export default Header;
