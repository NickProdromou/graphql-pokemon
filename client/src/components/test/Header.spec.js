import React from "react";
import { shallow } from "enzyme";
import Header from "../Header";
import { configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe("Header Component", () => {
  test("It renders without error", () => {
    const mockContext = { router: () => {} };
    const component = shallow(<Header />, { context: mockContext });

    expect(toJson(component)).toMatchSnapshot();
  });
});
