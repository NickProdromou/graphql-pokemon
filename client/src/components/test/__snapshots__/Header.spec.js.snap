// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component It renders without error 1`] = `
<header
  className="Header sc-ifAKCX ivpimK"
>
  <div
    className="sc-bxivhb bIAnAi"
  >
    <section
      className="sc-bwzfXH irrVft"
    >
      <div
        className="sc-htpNat bKOCBb"
      >
        <h1
          className="HeaderTitle"
        >
          Compare a pok√©mon
        </h1>
      </div>
    </section>
  </div>
</header>
`;
